mkosi/__init__.py

```diff
@@ -5086,15 +5086,24 @@ def run_verb(args: Args, images: Sequence[Config], *, resources: Path) -> None:
     if args.verb.needs_build():
         if args.verb != Verb.build and not args.force and not output.exists():
             die(
-                f"Image '{last.name()}' has not been built yet",
+                f"Image '{last.image}' has not been built yet",
                 hint="Make sure to build the image first with 'mkosi build' or use '--force'",
             )
 
         if not last.repart_offline and os.getuid() != 0:
-            die(f"Must be root to build {last.name()} image configured with RepartOffline=no")
+            die(f"Must be root to build {last.image} image configured with RepartOffline=no")
 
         check_workspace_directory(last)
 
+        if last.incremental:
+            for a, b in itertools.permutations(images, 2):
```

Comment:
Nit: After checking `a == b` we don't need to check `b == a`
```suggestion
            for a, b in itertools.combinations(images, 2):
```